Back out "Revert D29687143: [3/N] Nnapi Backend Delegate Preprocess: Basic OSS Test"

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61878

### Summary:
**CMakeLists.txt**
Android NNAPI delegate library was moved from test/cpp/jit/CMakeLists.txt to torch/CMakeLists.txt. This resolves the issue the original PR had, where the NNAPI delegate library was added to builds without Python (when it depends on Python).
**Original PR**: https://github.com/pytorch/pytorch/pull/61594

There's an error where the library cannot be built on MacOS. This problem existed in the original PR as well, but now an issue has been created: https://github.com/pytorch/pytorch/issues/61930

**test_backend_nnapi.py**
Also changed the skip unit test headers so that it's a little cleaner. Now the unit tests are skipped if the Nnapi delegate library file is not found. Previously, the skip was based on the platform (only allowing Linux).

### Test Plan:
Ran NNAPI delegate unit tests with `python test/test_jit.py TestNnapiBackend`

Reviewers:

Subscribers:
myuan

Tasks:

Tags:

Differential Revision: [D29799895](https://our.internmc.facebook.com/intern/diff/D29799895)

Commit:
Test PR: Back out "Revert D29687143: [3/N] Nnapi Backend Delegate Preprocess: Basic OSS Test"

Summary:
Using this PR to test CI tests in
https://github.com/pytorch/pytorch/pull/61787, since I'm also trying to
learn how to debug by rerunning jobs with SSH.

Test Plan:

Reviewers:

Subscribers:
myuan

Tasks:

Tags:

[ghstack-poisoned]

iseeyuan commented:
Looks like it has the undefined python symbol issue with Mac build?

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29799895).

Commit:
Update on "Test PR: Back out "Revert D29687143: [3/N] Nnapi Backend Delegate Preprocess: Basic OSS Test""


Summary:
Using this PR to test CI tests in
https://github.com/pytorch/pytorch/pull/61787, since I'm also trying to
learn how to debug by rerunning jobs with SSH.

Test Plan:
To run NNAPI delegate unit tests: `python test/test_jit.py TestNnapiBackend`

Reviewers:

Subscribers:
myuan

Tasks:

Tags:

Differential Revision: [D29799895](https://our.internmc.facebook.com/intern/diff/D29799895)

[ghstack-poisoned]

iseeyuan commented on test/jit/test_backend_nnapi.py:
```diff
@@ -0,0 +1,104 @@
+import torch._C
+from pathlib import Path
+
+from torch.testing._internal.common_utils import (
```
Please fix the lint error before landing it.

iseeyuan reviewed (approved):
LGTM. Thanks!

Commit:
Update on "Back out "Revert D29687143: [3/N] Nnapi Backend Delegate Preprocess: Basic OSS Test""


Summary:
Android NNAPI delegate library was moved from test/cpp/jit/CMakeLists.txt to torch/CMakeLists.txt. This resolves the issue the original PR had, where the NNAPI delegate library was added to builds without Python (when it depends on Python).
Original PR https://github.com/pytorch/pytorch/pull/61594

Test Plan:
Ran NNAPI delegate unit tests with `python test/test_jit.py TestNnapiBackend`

Reviewers:

Subscribers:
myuan

Tasks:

Tags:

Differential Revision: [D29799895](https://our.internmc.facebook.com/intern/diff/D29799895)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29799895).

Commit:
Update on "Back out "Revert D29687143: [3/N] Nnapi Backend Delegate Preprocess: Basic OSS Test""


### Summary:
**CMakeLists.txt**
Android NNAPI delegate library was moved from test/cpp/jit/CMakeLists.txt to torch/CMakeLists.txt. This resolves the issue the original PR had, where the NNAPI delegate library was added to builds without Python (when it depends on Python).
**Original PR**: https://github.com/pytorch/pytorch/pull/61594

There's an error where the library cannot be built on MacOS. This problem existed in the original PR as well, but now an issue has been created: https://github.com/pytorch/pytorch/issues/61930

**test_backend_nnapi.py**
Also changed the skip unit test headers so that it's a little cleaner. Now the unit tests are skipped if the Nnapi delegate library file is not found. Previously, the skip was based on the platform (only allowing Linux).

### Test Plan:
Ran NNAPI delegate unit tests with `python test/test_jit.py TestNnapiBackend`

Reviewers:

Subscribers:
myuan

Tasks:

Tags:

Differential Revision: [D29799895](https://our.internmc.facebook.com/intern/diff/D29799895)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29799895).
